internal/commands/app/create_test.go

```diff
@@ -310,6 +316,82 @@ func TestAppCreateHandler(t *testing.T) {
 		}, createdApp)
 	})
 
+	t.Run("when a template is not provided should prompt for template selection", func(t *testing.T) {
+		profile, teardown := mock.NewProfileFromTmpDir(t, "app_create_test")
+		defer teardown()
+		profile.SetRealmBaseURL("http://localhost:8080")
+
+		procedure := func(c *expect.Console) {
+			c.ExpectString("Please select a template from the available options")
+			c.SendLine("palm-pilot.bitcoin-miner")
+			c.ExpectEOF()
+		}
+
+		// TODO(REALMC-8264): Mock console in tests does not behave as initially expected
+		_, console, _, ui, consoleErr := mock.NewVT10XConsole()
+		assert.Nil(t, consoleErr)
+		defer console.Close()
+
+		doneCh := make(chan (struct{}))
+		go func() {
+			defer close(doneCh)
+			procedure(console)
+		}()
+
+		var createdApp realm.App
+		rc := mock.RealmClient{}
+		rc.CreateAppFn = func(groupID, name string, meta realm.AppMeta) (realm.App, error) {
+			createdApp = realm.App{
+				GroupID:     groupID,
+				ID:          "456",
+				ClientAppID: name + "-abcde",
+				Name:        name,
+				AppMeta:     meta,
+			}
+			return createdApp, nil
+		}
+		rc.ImportFn = func(groupID, appID string, appData interface{}) error {
+			return nil
+		}
+		rc.TemplatesFn = func() ([]realm.Template, error) {
+			return []realm.Template{
+				{
+					ID:   "palm-pilot.bitcoin-miner",
+					Name: "Mine bitcoin on your Palm Pilot from the comfort of your home, electricity not included",
+				},
+				{
+					ID:   "blackberry.important-business-app",
+					Name: "Oh wow, a Blackberry... you must a very powerful, extravagant man.",
+				},
+			}, nil
+		}
+
+		clientZipPkg, err := zip.OpenReader("testdata/react-native.zip")
+		assert.Nil(t, err)
+		rc.ClientTemplateFn = func(groupID, appID, templateID string) (*zip.Reader, error) {
+			return &clientZipPkg.Reader, nil
+		}
+		ac := mock.AtlasClient{}
+		ac.GroupsFn = func() ([]atlas.Group, error) {
+			return []atlas.Group{{ID: "123"}}, nil
+		}
+
+		cmd := &CommandCreate{createInputs{newAppInputs: newAppInputs{
+			Name:            "template-app",
+			Location:        realm.LocationVirginia,
+			DeploymentModel: realm.DeploymentModelGlobal,
+			ConfigVersion:   realm.DefaultAppConfigVersion,
+		}}}
+
+		assert.Nil(t, cmd.Handler(profile, ui, cli.Clients{Realm: rc, Atlas: ac}))
+
+		console.Tty().Close() // flush the writers
+		<-doneCh              // wait for procedure to complete
+
+		_, err = local.LoadApp(filepath.Join(profile.WorkingDirectory, cmd.inputs.Name))
```

Comment:
is this meant to prove the app was written?  if so, you'll want an assert on `appLocal.RootDir` to make sure its not blank as well here